### ISSUE TITLE
Aliyun support

### ISSUE THREAD

**Issue by LilyFaFa** [kubernetes/kops#4127]
Do you have plans to make kops support Aliyun? We plan to carry out the relevant work, do you have any suggestions? @justinsb

**Comment by chrislovecnm** [comment]
/asign @geojaz 

You should probably coordinate

**Comment by geojaz** [comment]
@LilyFaFa I'm going to be talking about this during office hours tomorrow 9 california time. If you'd like to join and talk about plans to coordinate, we would love to have you. https://github.com/kubernetes/kops#office-hours

**Comment by LilyFaFa** [comment]
@geojaz Thank you for your invitation, my partners and Ali engineers will attend the meeting.

**Comment by geojaz** [comment]
@LilyFaFa Could you cc your team here so we can organize the Aliyun implementation?

**Comment by xh4n3** [comment]
Hi @geojaz ,
We're currently reading the kops codes, now this seems really large for two or three weeks' development.
We plan to follow the process like DO does #2150  , to implement AliCloud as cloud provider.

Here are some raw ideas:
1. DNS Provider seems does the job for end-users to access clusters, not compulsory. The gossip-based thing should do the work for node to discover masters. So we can skip the implementation of dns provider.
2. We plan to port state store to AliCloud's OSS(object store) as an alternative option. 

Could you please provide some suggestions? Do we miss anything? Or, do you have a detailed roadmap for us to organize our work? 

Big thanks!

**Comment by LilyFaFa** [comment]
We started this work recently, my partners @xh4n3 @Xiao An 
Hope to get your help, thanks! @geojaz

**Comment by xlgao-zju** [comment]
I'd love to see this feature. 😆

**Comment by geojaz** [comment]
@LilyFaFa and other Aliyun folks, Do you have a list/issues for the remaining tasks outstanding to call this "done"? 

If we can get a solid list of concise features/bugs that need to be implemented/fixed, I think the community can help push this across the line and get aliyun support GA. Please let us know what we can do to make this happen!

**Comment by ebabani** [comment]
I feel like this is pretty close. When I tried it there were some client side checks that were failing since support for this cloud provider was not fully there.

I heard there are no CI integrations for alicloud yet. Feels like those are a requirement before this could be considered done, otherwise it would go out of date quick. What's required to get those in place?

I can help a little with this but most likely not before October.

**Comment by mtianyan** [comment]
I hope I can get it soon! Thanks everyone!

**Comment by simoniong** [comment]
can't wait to see it happen!

**Comment by geojaz** [comment]
Sounds good ; for now, let's try to focus on exactly _what_ needs to get done and we'll figure out how it gets done once we have those tasks. Once we have a list of reasonably concise outstanding features/issues and great issue/feature descriptions- the community will step in and help out. There are many people who really want to get started contributing to kops/kubernetes and if we can get a basic design, I think getting contributions to this will be fairly straightforward.

**Comment by legal90** [comment]
/remove-lifecycle stale

Hi guys! Are there any updates on this? Is there any way to use `kops` with Aliyun (from a fork maybe)?

**Comment by vousmeevoyez** [comment]
is the kops now supported on aliyun?

**Comment by bittopaz** [comment]
Hello @LilyFaFa, can you give an update on this, please?

**Comment by bittopaz** [comment]
I could get a successful output with the commands below:

```sh
export ALIYUN_ACCESS_KEY_ID=xxxxx
export ALIYUN_ACCESS_KEY_SECRET=xxxxx
export OSS_REGION=oss-cn-shanghai
export KOPS_FEATURE_FLAGS="AlphaAllowALI,+SpecOverrideFlag"

kops create cluster --cloud=alicloud --name=mycluster-3.k8s.local --networking=flannel --zones=cn-shanghai-b --ssh-public-key=~/.ssh/id_rsa.pub --override 'cluster.spec.etcdClusters[*].provider=Legacy'
```

Since I didn't use `--yes`, so it didn't create a cluster yet. Hopefully this could be useful to anyone that is also exploring on this topic.

**Comment by Ethyling** [comment]
Hey, I would like to deploy a k8s cluster on Alibaba cloud. Is there a chance that I can use Kops currently or should I use another tool (I would appreciate if you could give me some hints on this part)? I don't want to use a managed k8s cluster. Thank you!

**Comment by bittopaz** [comment]
@Ethyling you can read the doc change in https://github.com/kubernetes/kops/pull/8275, hope it can help

**Comment by Ethyling** [comment]
@bittopaz Ty, do you have any ETA regarding this issue?

**Comment by bittopaz** [comment]
@Ethyling nope. But I’m already running production clusters with kops in Alicloud. I’ll keep improving features related with Alicloud.

**Comment by hakman** [comment]
As mentioned in the [kOps 1.20 release notes](https://kops.sigs.k8s.io/releases/1.20-notes/#deprecations), Aliyun/Alicloud support is being deprecated:
> Due to lack of maintainers, the Aliyun/Alibaba Cloud support has been deprecated. The current implementation will be left as-is until the implementation needs updates or otherwise becomes incompatible. At that point, it will be removed. We very much welcome anyone willing to contribute to this cloud provider.

If anyone is interested in contributing, please contact the maintainers in the #kops-dev Slack channel.

**Comment by bittopaz** [comment]
@hakman Thank you for following this up.

There is a task list that I came out with when I tried to develop Alicloud provider in this https://github.com/kubernetes/kops/issues/4127#issuecomment-534536277. 

Also, if anyone would like to pick it up, he/she can use this PR https://github.com/kubernetes/kops/pull/8275 as a reference for setting up development. 

If anyone is interested in contributing, I'd be more than glad to transfer my knowledge to him/her.
